### PR TITLE
Soft delete user

### PR DESCRIPTION
**Pull request by wstnmssr**

### CODE REVIEW

File: xGame/Java/xgame/src/test/java/com/xgame/restservice/UserService.java [hunk 1]
```diff
@@ -149,4 +168,76 @@ void login_invalidCredentials() {
...
+		var userView = userService.registerNewUser(credentials);
+
+		var user = userRepo.findById(userView.getId()).get();
+
+		userService.deactivateUser(user.getId());
+		user = userRepo.findById(user.getId()).get();
+
+		assertTrue(user.getIsDeleted());
+
+		userRepo.deleteById(user.getId());
```
Comment (tatunka) [inline xGame/Java/xgame/src/test/java/com/xgame/restservice/UserService.java]: surround with try catch. delete user in finally block.

File: xGame/Java/xgame/src/test/java/com/xgame/restservice/UserService.java [hunk 2]
```diff
@@ -149,4 +168,76 @@ void login_invalidCredentials() {
...
+		assertTrue(fuzzySearchUserIds.contains(userEntity3.getId()));
+		assertTrue(emailIds.contains(userEntity3.getId()));
+		
+		assertTrue(!fuzzySearchUserIds.contains(userEntity2.getId()));
+		assertTrue(!nicknameIds.contains(userEntity2.getId()));
+
+		// cleanup
+		userRepo.delete(userEntity1);
+		userRepo.delete(userEntity2);
+		userRepo.delete(userEntity3);
```
Comment (tatunka) [inline xGame/Java/xgame/src/test/java/com/xgame/restservice/UserService.java]: same as above

File: xGame/Java/xgame/src/test/java/com/xgame/restservice/UserService.java [hunk 3]
```diff
@@ -149,4 +168,76 @@ void login_invalidCredentials() {
...
+		var testPassword = "testPassword1";
+
+		var credentials = new UserCredentials(testNickname, testEmail, testPassword);
+		var user = userService.registerNewUser(credentials);
+		
+		userService.deactivateUser(user.getId());
+
+		Assertions.assertThrows(Exception.class, () -> userService.login(credentials));
+
+		userRepo.deleteById(user.getId());
```
Comment (tatunka) [inline xGame/Java/xgame/src/test/java/com/xgame/restservice/UserService.java]: same as above

**Review by tatunka — CHANGES_REQUESTED** [review]
Add try-catch-finally to unit tests.

**Review by tatunka — APPROVED** [review]

**Review by aguentchev — APPROVED** [review]

**Pull request merged**